doc/macro_utils_design.md

```diff
@@ -235,4 +235,11 @@ MU_DIFFERENT(0,1) => 2
 
 Internally `MU_DIFFERENT` proposes numbers starting with MU_COUNT_ARG..0 as candidates for the expression result.
 
-If one of these numbers is not equal to any of the arguments then `MU_DIFFERENT` will evaluate to this number.
\ No newline at end of file
+If one of these numbers is not equal to any of the arguments then `MU_DIFFERENT` will evaluate to this number.
+
+### STATIC_ASSERT
+```c
+STATIC_ASSERT(CONDITION)
+```
+
+`STATIC_ASSERT` expands to a declaration of a static array. If the condition is false, it shall expand to an array with -1 elements which would result an error at compile time. Otherwise, it shall expand to an array with 1 elements.
```

Comment:
> exp [](http://example.com/codeflow?start=15&length=4)

I think a better explanation would be that evaluates the condition `CONDITION` and produces a compiler error if the condition is false.